src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -31,13 +32,15 @@ public class BookController {
     @Operation(summary = "Get all books",
             description = "Retrieve some set of books using pagination")
     @GetMapping
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_USER')")
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.getAll(pageable);
     }
 
     @Operation(summary = "Get book by ID",
             description = "Retrieve a book by its id")
     @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_USER')")
```

Comment:
I am wondering if this endpoint could work for both roles even without this annotation.